### PR TITLE
fix: preserve directory context for Go module files 

### PR DESCRIPTION
**Pull request by yousfiSaad**
- Add _run_osv_directory() to scan Go projects by directory instead of lockfile
- Early detection of go.mod/go.sum files in _process_asset()
- Preserves access to go.sum and module structure
- Backward compatible: all other file types use existing logic

Fixes context loss issue where OSV agent couldn't access go.sum when scanning Go projects. Now uses directory scanning (osv-scanner <dir>) instead of lockfile scanning (osv-scanner --lockfile go.mod) for Go files.

Changes: +47 lines in agent/osv_agent.py

### CODE REVIEW

**Comment by yousfiSaad** [comment]
# ⚠️ Testing and Review Notice

## Limited Testing

This fix has been tested with:
- Basic Go project structure (go.mod in root)
- Single module projects
- OXO scan workflow integration

**However, this fix has NOT been thoroughly tested with:**
- Nested Go modules
- Relative path scenarios
- Edge cases with different file path formats
- All supported file types (backward compatibility)
- Various OXO runtime configurations

## Code Review Required
⚠️ **Needs comprehensive testing** - Limited test coverage  
🔍 **Requires code review** - Before merging to main

File: agent/osv_agent.py [hunk 1]
```diff
@@ -184,6 +184,37 @@ def _run_osv(path: str, content: bytes) -> str | None:
...
+        file_path.write_text(patched_content.decode("utf-8", errors="ignore"), encoding="utf-8")
+        
+        # Scan directory instead of just the file
+        command = ["/usr/local/bin/osv-scanner", "--format", "json", str(module_dir)]
+        output = subprocess.run(command, cwd=module_dir, capture_output=True, text=True, check=False)
+        
+        if _is_valid_osv_result(output.stdout):
+            return output.stdout
+        return None
+    except Exception as e:
```
Comment (3asm) [inline agent/osv_agent.py]: Catching all Exceptions is something we prohibit and commonly prohibited in code style guides (see Google style guide as example: https://google.github.io/styleguide/pyguide.html).

This can often hide bugs that gets unnoticed.

File: agent/osv_agent.py [hunk 2]
```diff
@@ -302,6 +333,22 @@ def _process_asset(self, message: m.Message) -> None:
         if file_type in FILE_TYPE_BLACKLIST:
             logger.debug("File type is blacklisted.")
             return
+        
+        # Special handling for Go module files to preserve directory context
+        if path and path.endswith(("go.mod", "go.sum")):
```
Comment (3asm) [inline agent/osv_agent.py]: Prefer explicit checks, like path is not None. Also common source of bugs.

File: agent/osv_agent.py [hunk 3]
```diff
@@ -302,6 +333,22 @@ def _process_asset(self, message: m.Message) -> None:
...
+            scan_results = _run_osv_directory(path, content)
+            if scan_results is not None:
+                parsed_output = osv_output_handler.parse_osv_output(
+                    scan_results, self.api_key
+                )
+                if len(parsed_output) > 0:
+                    vulnerability_location = _prepare_vulnerability_location(message)
+                    self._emit_vulnerabilities(
+                        output=parsed_output,
+                        vulnerability_location=vulnerability_location,
```
Comment (3asm) [inline agent/osv_agent.py:354]: The code needs unit tests 😃 .

**Review by 3asm — CHANGES_REQUESTED** [review]
Thanks for the contribution. The code looks good, it does however need unit test to make sure it is both working and refactorable in the future.

File: agent/osv_agent.py (hunk 1 above)
Comment (yasser-abassi) [reply agent/osv_agent.py]: fixed

**Review by 3asm — APPROVED** [review]
